streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -634,6 +636,25 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         this(internalTopologyBuilder, config, clientSupplier, Time.SYSTEM);
     }
 
+    @SuppressWarnings("unchecked")
+    private void configureSerDe(final Set<SinkNode> sinks, final Set<SourceNode> sources) {
```

Comment:
Besides source and sink nodes, there are other places in the middle of the topology that would pass in serdes:

1. Materialized.
2. Grouped.
3. Joined.

We need to cover those cases as well.

Reply:
w.r.t. Materialized, I was looking at StreamsBuilder:
```
        final ConsumedInternal<K, V> consumedInternal =
                new ConsumedInternal<>(Consumed.with(materializedInternal.keySerde(), materializedInternal.valueSerde()));
```
which leads to:
```
    public ConsumedInternal(final Serde<K> keySerde,
                            final Serde<V> valSerde,
                            final TimestampExtractor timestampExtractor,
                            final Topology.AutoOffsetReset offsetReset) {
        this(Consumed.with(keySerde, valSerde, timestampExtractor, offsetReset));
```
Should configure() be called in the above place ?

Reply:
Good question.

I think this boils down to the question, if auto-configuration should be a DSL or PAPI feature. My personal take it, it should be a PAPI feature. What do you think @guozhangwang 

If it's a PAPI feature, calling `configure()` in `ConsumedInternal()` seem not to be necessary.